numpyro/contrib/funsor/enum_messenger.py

```diff
@@ -501,6 +501,8 @@ def process_message(self, msg):
         if msg["type"] != "sample" or \
                 msg.get("done", False) or msg["is_observed"] or msg["infer"].get("expand", False) or \
                 msg["infer"].get("enumerate") != "parallel" or (not msg["fn"].has_enumerate_support):
+            if msg["type"] == "control_flow":
+                msg["kwargs"]["enum"] = True
```

Comment:
nit: this should probably live in `infer_config`

Reply:
Hmm, I need `scan_wrapper` knows that `enum=True` but `scan_wrapper` only accesses to `args`, `kwargs`, not `infer` field.

Reply:
Sorry, I meant that this logic setting `msg["kwargs"]["enum"] = True` should live in `infer_config`, not in `enum`.

Reply:
Do you suggest that when we use `config_enumerate(fn)`, we will add `enum` kwargs to `control_flow` msg? Currently, `infer_config` does
```
    if msg["type"] in ("sample", "param"):
            msg["infer"].update(self.config_fn(msg))
```
so we need to change it to
```
        if msg["type"] in ("sample", "param"):
            msg["infer"].update(self.config_fn(msg))
        if msg["type"] == "control_flow":
         msg["kwargs"].update(self.config_fn(msg))
```
and in `config_enumerate`, we need to check if `msg` is `control_flow` to return `{"enum": True}`. It is a bit verbose to me and violate the purpose of `infer_config`. In addition, we don't need to use `config_enumerate` if users already provide `infer={"enumerate": "parallel"}` in `sample` statements. In that case, the only handler that affects `control_flow` is `enum`.

Reply:
> In that case, the only handler that affects control_flow is enum.

Ah, yes, good point, I guess what you have here is fine.